integration/presto/src/test/scala/org/apache/carbondata/presto/integrationtest/PrestoInsertIntoTableTestCase.scala

```diff
@@ -0,0 +1,207 @@
+      .executeQuery("select count(*) AS RESULT from testdb.testtable")
+    val expectedResult1: List[Map[String, Any]] = List(Map("RESULT" -> 4))
+    assert(actualResult1.equals(expectedResult1))
+    // filter query
+    val actualResult2: List[Map[String, Any]] = prestoServer
+      .executeQuery("select count(*) AS RESULT from testdb.testtable WHERE dob = timestamp '1998-12-16 10:12:09'")
+    val expectedResult2: List[Map[String, Any]] = List(Map("RESULT" -> 2))
+    assert(actualResult2.equals(expectedResult2))
+  }
+
+  test("test if the table status contains the segment file name for each load") {
+    prestoServer.execute("insert into testdb.testtable values(10, current_date, 'INDIA', 'Chandler', 'qwerty', 'usn20392',10000.0,16.234567,25.678,timestamp '1994-06-14 05:00:09',smallint '23', true)")
+    val absoluteTableIdentifier: AbsoluteTableIdentifier = getAbsoluteIdentifier("testdb", "testtable")
+    val carbonTable = SchemaReader.readCarbonTableFromStore(absoluteTableIdentifier)
+    val ssm = new SegmentStatusManager(carbonTable.getAbsoluteTableIdentifier)
+    ssm.getValidAndInvalidSegments.getValidSegments.asScala.foreach { segment =>
+      val loadMetadataDetails = segment.getLoadMetadataDetails
+      assert(loadMetadataDetails.getSegmentFile != null)
+    }
+  }
+
+  test("test for query when insert in progress") {
+    prestoServer.execute("insert into testdb.testtable values(10, current_date, 'INDIA', 'Chandler', 'qwerty', 'usn20392',10000.0,16.234567,25.678,timestamp '1994-06-14 05:00:09',smallint '23', true)")
+    val query = "insert into testdb.testtable values(10, current_date, 'INDIA', 'Chandler', 'qwerty', 'usn20392',10000.0,16.234567,25.678,timestamp '1994-06-14 05:00:09',smallint '23', true)"
+    val asyncQuery = runSqlAsync(query)
+    val actualResult1: List[Map[String, Any]] = prestoServer.executeQuery("select count(*) AS RESULT from testdb.testtable WHERE dob = timestamp '1994-06-14 05:00:09'")
+    val expectedResult1: List[Map[String, Any]] = List(Map("RESULT" -> 1))
+    assert(actualResult1.equals(expectedResult1))
+    assert(asyncQuery.get().equalsIgnoreCase("PASS"))
+    val actualResult2: List[Map[String, Any]] = prestoServer.executeQuery("select count(*) AS RESULT from testdb.testtable WHERE dob = timestamp '1994-06-14 05:00:09'")
+    val expectedResult2: List[Map[String, Any]] = List(Map("RESULT" -> 2))
+    assert(actualResult2.equals(expectedResult2))
+  }
+
+  class QueryTask(query: String) extends Callable[String] {
+    override def call(): String = {
+      var result = "PASS"
+      try {
+        prestoServer.execute(query)
+      } catch {
+        case ex: Exception =>
+          println(ex.printStackTrace())
+          result = "FAIL"
+      }
+      result
+    }
+  }
+
+  private def runSqlAsync(sql: String): Future[String] = {
+    val future = executorService.submit(
+      new QueryTask(sql)
+    )
+    Thread.sleep(2)
+    future
+  }
+
+  override def afterAll(): Unit = {
```

Comment:
Please update the documentaion mentioning the insert syntax, what and all supported and what and all open.

Reply:
In the cluster please test concurrent insert

Reply:
added in document, need to test in cluster about the concurrent test

Reply:
cluster test is done, and concurrent test also done. Since we dont have clean files scenarios and all , failure cases are not clearly handled like spark. This we can improvise in later part.